src/Rules/Comparison/StrictComparisonOfDifferentTypesRule.php

```diff
@@ -50,16 +48,7 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
-		if (
-			(!$leftType->accepts($rightType) && !$rightType->accepts($leftType))
-			|| (
-				$leftType instanceof IntegerType
-				&& $rightType instanceof FloatType
-			) || (
-				$rightType instanceof IntegerType
-				&& $leftType instanceof FloatType
-			)
-		) {
+		if ($leftType->isSupersetOf($rightType)->no()) {
```

Comment:
The same could be applied to ImpossibleInstanceOfRule.

Reply:
Good idea. I've just tried that but there is a minor issue. It works a lot better that it used to a has found a 14 places in phpstan coded base where instanceof always evaluates to true. For example the it can now see that the highlighted condition is always true

![image](https://user-images.githubusercontent.com/175109/30188870-2d9ee658-9431-11e7-8b1b-a4eeec5846b9.png)


I'm afraid that the only reasonable solution is to emit error only when instanceof always evaluates to false, but not when instanceof always evaluates to true.

![image](https://user-images.githubusercontent.com/175109/30188733-654edf32-9430-11e7-90d6-b0c5af4c6cb4.png)

Reply:
@JanTvrdik Do you think that some places that report `instanceof always evaluates to true` are invalid? I checked the first one from your screenshot and there's a documentation issue - this isn't true: `@param \PhpParser\Node[] $nodes`.